File: python/69. Sqrt(x).py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        if x == 0:
            return 0
        low = 0
        high = x
        while True:
            mid = (low + high) / 2
            if mid * mid > x:
                high = mid - 1
            else:
                if (mid + 1) ** 2 > x:
                    return mid
                low = mid + 1
            

File: cpp/51-100/69. Sqrt(x).cpp
class Solution {
public:
	int mySqrt(long x) {
		long low = 0, high = x;
		long mid;
		while(low <= high)
		{
			mid = (low + high) / 2;
			if(mid * mid <= x && (mid + 1) * (mid + 1) > x)
				return mid;
			else
				if(mid * mid > x)
					high = mid - 1;
				else
					low = mid + 1;
		}
		return mid;
	}
};